containers/src/Data/Set.hs

```diff
@@ -136,6 +136,12 @@ module Data.Set (
             -- * Map
             , S.map
             , mapMonotonic
+#if __GLASGOW_HASKELL__ >= 710
```

Comment:
This is quite confusing and annoying; please avoid APIs exporting things conditional on GHC or `base` versions

Reply:
@hvr, don't worry; I won't let it through like that. Please join the discussion on the libraries list about whether to add some of these.

Reply:
@hvr there are two problems with that. 7.6 and 7.8 don't have Applicative constraint on Monad class and 7.6 does not have fromListN at all, which would result in two different implementations of the same function having different performance characteristics. How do you propose to solve it?

Reply:
ping

Reply:
ping

Reply:
@hvr ping

Reply:
As there is no response from @hvr can you @treeowl give a definite list of required changes that need to be made in order for this to be merged, including how you want multiple GHC support to happen?